File: day1/part1.py
#!/usr/bin/env python3

import sys
fname = "input"
with open(fname, "rt") as file:
    lines = file.readlines()

numbers = [int(x) for x in lines]

for i in numbers:
	for j in numbers:
		if (i + j == 2020):
			print ("%s %s %s %s" % (i, j, i + j, i*j))

File: day13/part1.js
#!/usr/bin/env node

import { readLines } from '../common/readers.js';

async function main() {	

	const lines = await readLines('input');
	const now = + lines[0];
	const buses = lines[1].split(',').filter(v => v !== 'x').map(d => +d);
	console.log({ now });
	let min = now;
	let minBus = 0;
	for (const bus of buses) {
		let round = Math.floor(now / bus);
		let nextDeparture = bus * round;
		if (nextDeparture < now) nextDeparture += bus;
		let wait = nextDeparture - now;
		if (wait < min) {
			min = wait;
			minBus = bus;
		}
		console.log(bus, round, nextDeparture, wait);
	}
	console.log('Next bus: ', min, minBus, 'Answer:', min * minBus);
}

main();
